tools/rc/usResourceCompiler.cpp

```diff
@@ -384,144 +432,95 @@ static int us_zip_writer_add_from_zip_reader(mz_zip_archive *pZip, mz_zip_archiv
 int main(int argc, char** argv)
 {
   int compressionLevel = 6;
-  int argIndex = 0;
-  int bPrintHelp = 0;
-  int bAppendMode = 0;
-
-  int errCode = US_OK;
-
-  int mergeFlag = 0;
-  const char* zipFile = NULL;
-  const char* bundleName = NULL;
-  size_t bundleNameLength = 0;
-
-  FILE* zipfileStream = NULL;
-  mz_zip_archive writeArchive;
-
-  us_archived_names archivedNames;
-  us_archived_names archivedDirs;
-
-  FILE* appendStream = NULL;
-
-  char archiveName[MZ_ZIP_MAX_ARCHIVE_FILENAME_SIZE];
-
-  int numZipArgs = 0;
-  int* zipArgIndices = NULL;
-
-  mz_zip_archive currFileArchive;
-
-  int zipArgIndex = 0;
-
-  char readBuffer[1024];
-  mz_uint numRead = 0;
-
-
-  // ---------------------------------------------------------------------------------
-  //      COMMAND LINE VALIDATION
-  // ---------------------------------------------------------------------------------
+  int errCode = US_OK, return_code = EXIT_SUCCESS;
+  std::string zipFile;
+  std::string bundleName;
+  std::string archiveName;
+
+  argc -= (argc > 0);
+  argv += (argc > 0); // skip program name argv[0] if present
```

Comment:
Isn't the program name always present as the first element in argv? When is it not present?